sqlx-sqlite/src/connection/establish.rs

```diff
@@ -92,10 +92,22 @@ impl EstablishParams {
         }
 
         if !query_params.is_empty() {
+            let query_string = query_params
+                .iter()
+                .map(|(key, value)| {
+                    format!(
+                        "{}={}",
+                        percent_encoding::percent_encode(key.as_bytes(), NON_ALPHANUMERIC),
+                        percent_encoding::percent_encode(value.as_bytes(), NON_ALPHANUMERIC)
+                    )
+                })
+                .collect::<Vec<_>>()
+                .join("&");
```

Comment:
Instead, you could use the `form_urlencoded` crate which is already a transitive dependency via `url`:

```rust
let query_string = form_urlencoded::Serializer::new(String::new())
    .extend_pairs(query_params)
    .finish();
```